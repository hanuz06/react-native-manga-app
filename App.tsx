import React, { useState } from "react";
import { SafeAreaProvider } from "react-native-safe-area-context";
import { StyleSheet } from "react-native";
import { Provider as PaperProvider, DefaultTheme } from "react-native-paper";

// This import loads the firebase namespace.
import firebase from "firebase/app";

// These imports load individual services into the firebase namespace.
import "firebase/storage";
import "firebase/database";

import { createStore, combineReducers, applyMiddleware } from "redux";
import { Provider as ReduxProvider } from "react-redux";
import ReduxThunk from "redux-thunk";
import mangaReducer from "./store/reducers/mangaReducer";
import { NavigationContainer } from "@react-navigation/native";
import { RootNavigator } from "./navigation/MangaAppNavigation";
import { KEY } from "react-native-dotenv";

const rootReducer = combineReducers({
  manga: mangaReducer,
});

const store = createStore(rootReducer, applyMiddleware(ReduxThunk));

const theme = {
  ...DefaultTheme,
  roundness: 2,
  colors: {
    ...DefaultTheme.colors,
    surface: "#FFFFF0",
    primary: "#d32f2f",
    accent: "white",
  },
};

var firebaseConfig = {
  apiKey: { KEY },
  authDomain: "react-native-manga-reader-app.firebaseapp.com",
  databaseURL: "https://react-native-manga-reader-app.firebaseio.com",
  projectId: "react-native-manga-reader-app",
  storageBucket: "react-native-manga-reader-app.appspot.com",
  messagingSenderId: "863581223721",
  appId: "1:863581223721:web:1104e2847f181021abd538",
  measurementId: "G-9JCXKZ66LR",
};
// Initialize Firebase
firebase.initializeApp(firebaseConfig);

export default function App() {
  const [fontLoaded, setFontLoaded] = useState<boolean>(false);

  return (
    <SafeAreaProvider>
      <ReduxProvider store={store}>
        <PaperProvider theme={theme}>
          <NavigationContainer>
            <RootNavigator />
          </NavigationContainer>
        </PaperProvider>
      </ReduxProvider>
    </SafeAreaProvider>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
