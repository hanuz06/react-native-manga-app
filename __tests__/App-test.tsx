import "react-native";
import React from "react";
import App from "../App";

import renderer from "react-test-renderer";

describe("<App />", () => {
  it("match to snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
    // expect(tree).not.toEqual(null);
  });
});

it("renders correctly", () => {
  renderer.create(<App />);
});

// it('works', () => {
//   expect(1).toBe(1);
// });
