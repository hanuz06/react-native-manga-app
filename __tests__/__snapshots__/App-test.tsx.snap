// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> match to snapshot 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
